pkg/security/resolvers/tags/resolver_linux.go

```diff
@@ -46,35 +54,46 @@ func (t *LinuxResolver) Start(ctx context.Context) error {
 			case <-ctx.Done():
 				return
 			case <-delayerTick.C:
-				select {
-				case workload := <-t.workloadsWithoutTags:
-					t.checkTags(workload)
-				default:
-				}
 
+			WORKLOAD:
+				for {
+					select {
+					case workload := <-t.workloadsWithoutTags:
+						t.checkTags(workload)
+					default:
+						break WORKLOAD
+					}
+				}
 			}
 		}
 	}()
 
 	return nil
 }
 
+func needsTagsResolution(cgce *cgroupModel.CacheEntry) bool {
+	return len(cgce.ContainerID) != 0 && !cgce.WorkloadSelector.IsReady()
+}
+
 // checkTags checks if the tags of a workload were properly set
-func (t *LinuxResolver) checkTags(workload *cgroupModel.CacheEntry) {
-	// check if the workload tags were found
-	if workload.NeedsTagsResolution() {
-		// this is a container, try to resolve its tags now
-		if err := t.fetchTags(workload); err != nil || workload.NeedsTagsResolution() {
-			// push to the workloadsWithoutTags chan so that its tags can be resolved later
-			select {
-			case t.workloadsWithoutTags <- workload:
-			default:
+func (t *LinuxResolver) checkTags(pendingWorkload *pendingWorkload) {
+	workload := pendingWorkload.CacheEntry
+	// check if the workload tags were found or if it was deleted
+	if !workload.Deleted.Load() && needsTagsResolution(workload) {
+		// this is an alive cgroup, try to resolve its tags now
+		if err := t.fetchTags(workload); err != nil || needsTagsResolution(workload) {
+			if pendingWorkload.retries--; pendingWorkload.retries > 0 {
```

Comment:
Can we add an log message to indicate that we failed to resolve the tags of particular workload, either because the maximum number of retries was reached, or because the chan was full